Client/Application/AppDelegate.swift

```diff
@@ -89,16 +89,31 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject?) -> Bool {
-        if let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) where components.scheme == "firefox" && components.host == "open-url" {
-            if let query = components.query, item = components.queryItems?.first as? NSURLQueryItem {
-                if item.name == "url" && item.value != nil {
-                    if let newURL = NSURL(string: item.value!) {
-                        let tab = self.tabManager.addTab(request: NSURLRequest(URL: newURL))
-                        self.tabManager.selectTab(tab)
+        let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false)
+        if components!.scheme != "firefox" && components!.scheme != "firefox-x-callback" {
```

Comment:
Not sure when `components` can be nil here, but since the `url` is coming from an external app, we should be careful and avoid force unwrapping. Let's guard this with an `if let components = ...`.